lib/src/contracts/stream_handler.dart

```diff
@@ -12,6 +12,8 @@ abstract class StreamHandler {
   /// Add a listener to the event channel stream for pusher,
   /// any class that extends [StreamHandler] should use this method.
   void registerListener(String classId, dynamic Function(dynamic) method) {
+    _eventStreamSubscription?.cancel();
```

Comment:
Yes, connection state changes still work after binding because Channel and PusherClient extend this abstract class on their own. bind() and connect() only re-subscribes their own instance.